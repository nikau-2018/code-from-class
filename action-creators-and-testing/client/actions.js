export function addWombat (wombat) {
  return 'A string'
}

export function delWombat (wombat) {
  return {
    type: 'DEL_WOMBAT',
    wombat
  }
}

export function updateWombat (wombat, newWombat) {
  return {
    type: 'UPDATE_WOMBAT',
    newWombat,
    wombat
  }
}