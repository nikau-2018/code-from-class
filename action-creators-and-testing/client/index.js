import React from 'react'
import ReactDOM from 'react-dom'
import {createStore} from 'redux'

import App from './components/App'
import wombatReducer from './reducer'

const store = createStore(wombatReducer,
  window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
)

document.addEventListener('DOMContentLoaded', () => {
  render()
  store.subscribe(render)
})

function render () {
  const state = store.getState()
  const wombats = state.wombats
  ReactDOM.render(
    <App dispatch={store.dispatch} wombats={wombats} />,
    document.getElementById('app')
  )
}
