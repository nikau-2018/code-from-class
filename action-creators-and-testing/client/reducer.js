const initialWombatState = {
  wombats: ['Gertrude', 'Bartholemew']
}

const wombatReducer = (state = initialWombatState, action) => {
  switch (action.type) {
    case 'ADD_WOMBAT':
      return {
        wombats: [...state.wombats, action.wombat]
      }

    case 'DEL_WOMBAT':
      return {
        wombats: state.wombats.filter((wombat) => wombat !== action.wombat)
      }

    case 'UPDATE_WOMBAT':
      return {
        wombats: [
          ...state.wombats.filter(wombat => wombat !== action.wombat),
          action.newWombat
        ]
      }

    default:
      return state
  }
}

export default wombatReducer
