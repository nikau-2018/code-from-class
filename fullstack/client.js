const request = require('superagent')

console.log('making the request')

request.get('http://localhost:3000/api/v1/users')
  .then(res => {
    // console.log('res body:', res.body)
    res.body.users.forEach(user => {
      console.log(user.name)
    })
  })
  .catch(err => {
    console.error(err)
  })
