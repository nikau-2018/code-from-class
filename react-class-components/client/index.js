import React from 'react'
import ReactDOM from 'react-dom'

import HelloTemplate from './Hello'

const data = {
  name: 'humans'
}

const placeToMount = document.getElementById('root')

ReactDOM.render(<HelloTemplate name={data.name} />, placeToMount)
