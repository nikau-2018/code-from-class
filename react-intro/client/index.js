const React = require('react')
const ReactDOM = require('react-dom')

function ShowDog (props) {
  const { dogUrl, label } = props
  return (
    <div>
      <p>{label.length > 0 ? label : 'NO LABEL SPECIFIED'}</p>
      <img src={dogUrl} />
    </div>
  )
}

function ShowDate () {
  const date = new Date()

  return (
    <p>{`The date is: ${date}`}</p>
  )
}

function helloTemplate (props) {
  return (
    <div>
      <div>hello {props.name}</div>
      <ShowDog label='' dogUrl={data.dogUrl} />
      <ShowDate />
    </div>
  )
}

const data = {
  dogUrl: 'http://www.placepuppy.net/400/250',
  name: 'humans'

}
const view = helloTemplate(data)

const placeToMount = document.getElementById('root')

ReactDOM.render(view, placeToMount)
