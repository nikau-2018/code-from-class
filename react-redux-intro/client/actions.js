export const ADD_COLOR = 'ADD_COLOR'
export const ADD_USER = 'ADD_USER'

export function addColor (color) {
  return {
    type: ADD_COLOR,
    color: color
  }
}

export function addUser (userName) {
  return {
    type: ADD_USER,
    userName: userName
  }
}
