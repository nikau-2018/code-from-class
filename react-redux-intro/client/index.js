import React from 'react'
import ReactDOM from 'react-dom'
import {createStore} from 'redux'
import {Provider} from 'react-redux'

import App from './components/App'
import reducer from './reducers'

const store = createStore(
  reducer,
  window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
)

document.addEventListener('DOMContentLoaded', () => {
  ReactDOM.render(
    <Provider store={store}>
      <App />
    </Provider>,
    document.getElementById('app')
  )
})
