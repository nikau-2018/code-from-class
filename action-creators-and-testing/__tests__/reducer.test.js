import reducer from '../client/reducer'
import { addWombat } from '../client/actions'

test('It works', () => {
  expect(true).toBe(true)
})

const initialState = {
  wombats: []
}

test('Wombats is an array after ADD_WOMBAT', () => {
  const action = {
    type: 'ADD_WOMBAT',
    wombat: 'Foo'
  }
  const expected = true
  const actual = reducer(initialState, action)
  expect(Array.isArray(actual.wombats)).toBe(expected)
})

test('ADD_WOMBAT adds a wombat', () => {
  const expected = 'Foo'
  const action = {
    type: 'ADD_WOMBAT',
    wombat: expected
  }
  const actual = reducer(initialState, action).wombats
  expect(actual).toContain(expected)
})

test('addWombat matches the last snapshot', () => {
  expect(addWombat('Hi')).toMatchSnapshot()
})