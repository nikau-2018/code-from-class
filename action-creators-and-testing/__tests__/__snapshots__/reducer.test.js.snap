// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addWombat matches the last snapshot 1`] = `"A string"`;
